include/rcutils/shared_library.h

```diff
@@ -0,0 +1,149 @@
+ * rcutils_load_shared_library(&bar, "library_name", rcutils_get_default_allocator()); // ok
+ * void * symbol = rcutils_get_symbol(&bar, "bazinga"); // ok
+ * bool is_bazinga_symbol = rcutils_has_symbol(&bar, "bazinga"); // ok
+ * rcutils_ret_t ret = rcutils_unload_shared_library(&bar); // ok
+ * if (ret != RCUTILS_RET_ERROR) {
+ *   // error handling
+ * }
+ * ```
+ * */
+RCUTILS_PUBLIC
+RCUTILS_WARN_UNUSED
+rcutils_shared_library_t
+rcutils_get_zero_initialized_shared_library(void);
+
+/// Return shared library pointer.
+/**
+ * \param[inout] lib struct with the shared library pointer and shared library path name
+ * \param[in] library_path string with the path of the library
+ * \param[in] allocator to be used to allocate and deallocate memory
+ * \return `RCUTILS_RET_OK` if successful, or
+ * \return `RCUTILS_RET_BAD_ALLOC` if memory allocation fails, or
+ * \return `RCUTILS_RET_ERROR` if an unknown error occurs, or
+ * \return `RCUTILS_RET_INVALID_ARGUMENT` for invalid arguments
+ */
+RCUTILS_PUBLIC
+RCUTILS_WARN_UNUSED
+rcutils_ret_t
+rcutils_load_shared_library(
+  rcutils_shared_library_t * lib,
+  const char * library_path,
+  rcutils_allocator_t allocator);
+
+/// Return shared library symbol pointer.
+/**
+ * \param[in] lib struct with the shared library pointer and shared library path name
+ * \param[in] symbol_name name of the symbol inside the shared library
+ * \return shared library symbol pointer, if the symbol doesn't exist then returns NULL.
+ */
+RCUTILS_PUBLIC
+RCUTILS_WARN_UNUSED
+void *
+rcutils_get_symbol(const rcutils_shared_library_t * lib, const char * symbol_name);
+
+/// Return true if the shared library contains a specific symbol name otherwise returns false.
+/**
+ * \param[in] lib struct with the shared library pointer and shared library path name
+ * \param[in] symbol_name name of the symbol inside the shared library
+ * \return if symbols exists returns true, otherwise returns false.
+ */
+RCUTILS_PUBLIC
+RCUTILS_WARN_UNUSED
+bool
+rcutils_has_symbol(const rcutils_shared_library_t * lib, const char * symbol_name);
+
+/// Unload the shared library.
+/**
+ * \param[in] lib rcutils_shared_library_t to be finalized
+ * \return `RCUTILS_RET_OK` if successful, or
+ * \return `RCUTILS_RET_INVALID_ARGUMENT` for invalid arguments, or
+ * \return `RCUTILS_RET_ERROR` if an unknown error occurs
+ */
+RCUTILS_PUBLIC
+RCUTILS_WARN_UNUSED
+rcutils_ret_t
+rcutils_unload_shared_library(rcutils_shared_library_t * lib);
+
+/// Get the library name for the compiled platform
+/**
+ * \param[in] library_name library base name (without prefix and extension)
+ * \param[out] library_name_platform library name for the compiled platform
+ * \return `RCUTILS_RET_OK` if successful, or
+ * \return `RCUTILS_RET_ERROR` if an unknown error occurs
+ */
+RCUTILS_PUBLIC
+RCUTILS_WARN_UNUSED
+rcutils_ret_t
+rcutils_get_platform_library_name(const char * library_name, char * library_name_platform);
```

Comment:
The API doesn't define how large `library_name_platform` is so this can easily result in writing into wrong memory.

Reply:
fixed https://github.com/ros2/rcutils/pull/215/commits/510bf355f0cee666d1c096075cae1e6b255babbd